furusiya/model/entities/party/player.py

```diff
@@ -48,14 +48,26 @@ def unmount(self, horse):
             self.mounted = False
             horse.is_mounted = False
 
+    def _rest(self, current_hp, max_hp):
+        return int(config.data.player.restingPercent * max_hp / current_hp)
+
     def rest(self):
-        if config.data.features.allowResting:
-            fighter = self.get_component(Fighter)
-            hp_gained = int(config.data.player.restingRatio * fighter.max_hp / fighter.hp)
-            fighter.heal(hp_gained)
-            return 'rested'
-        else:
-            return 'didnt-take-turn'
+        fighter = self.get_component(Fighter)
+        hp_gained = self._rest(fighter.hp, fighter.max_hp)
+        fighter.heal(hp_gained)
+        return 'rested'
+
+    def rest_to_full_health(self):
```

Comment:
Despite your comments, this code is still very convoluted and needs to be rewritten. The relationship between `rest` and `rest_to_full_health` is not clear, there is an interaction between these two.

You don't need `temp_health`. Simply calculate the number of turns as `(max_health - current_health) / health_gained_per_turn`. The latter should be a simple calculation of `max_health * rest_health_percent`.

Adding `self.rest() # rest for this turn` doesn't make sense. Even if you look at the code calling this code, it's not clear where the turn processing is taking place. 

This method isn't actually making you rest to full health. It's calculating and returning some number of turns.

Returning a string for a player move and a dictionary this one time, makes the code harder to understand. 

Most of this code does not seem testable.